### ISSUE TITLE
No response when pressing the export button to export data in CSV format using ImportExportTools NG add-on

### ISSUE THREAD

**Issue by marusan0303** [thunderbird/import-export-tools-ng#646]
Hello,
I am very grateful for the help you provided me before.
If possible, I would appreciate your assistance once again.

I recently updated Thunderbird to version 128.4.0esr (64-bit).

When I try to export data in CSV format using the ImportExportTools NG add-on, nothing happens when I press the export button.
Do you have any suggestions for how to resolve this issue?

Thank you very much.

**Comment by cleidigh** [comment]
@marusan0303 
Good to see you again.
I am not clear about the button you refer to. The export to csv is either on the right click menu in the message list or on a folder.
Can you clarify how you are exporting?
@cleidigh

**Comment by marusan0303** [comment]
Hello again,
I’m happy to be in touch with you once more.

Regarding exporting to CSV:
I right-click on the Inbox, then select "Search and Export Messages." Next, I set the search criteria by specifying the sent date, and finally, I click the export button. This should save the messages as an Excel file on the desktop.

However, when I press the export button, there is no response.

Thank you very much for your assistance.
![aaa](https://github.com/user-attachments/assets/b980b440-ece4-402f-9468-1079c2787761)

**Comment by st-schilling** [comment]
Hello @cleidigh ,

I have the same issue, the console prints:

```
JavaScript error: chrome://mboximport/content/mboximport/sdOverlay.js, line 121: NS_ERROR_XPC_BAD_CONVERT_JS: Could not convert JavaScript argument arg 0 [nsIFilePicker.init]
```

The issue seems to be more general, not directly related to CSV exports, but the file dialogue.

![image](https://github.com/user-attachments/assets/7b6f4232-1b6e-48bd-bbe0-219572ea4def)


It seems to be the same issue as https://github.com/RealRaven2000/QuickFolders/issues/476.

Thanks.
kniffte

**Comment by cleidigh** [comment]
@st-schilling 
Thanks, no one said before it was the search dialog, that's why I couldn't reproduce.
However, on trying to check I just discovered that the overlay of the search export is completely missing so they changed something with the dialog in 128.3.3. What version are you using?
@cleidigh

**Comment by st-schilling** [comment]
Hello @cleidigh ,

you're welcome :)

ImportExportTools NG v. 14.1.2:
![image](https://github.com/user-attachments/assets/09244aec-b904-4bc9-89eb-b30af08e22af)

Thunderbird 128.4.0esr (64-Bit) on Ubuntu:
![image](https://github.com/user-attachments/assets/6ffeac8b-3fc0-4b59-872a-52ee8a9d2d71)

Thanks.
kniffte

**Comment by cleidigh** [comment]
@st-schilling 
I am not seeing the export button anymore, are you?

**Comment by st-schilling** [comment]
@cleidigh ,
yes, please see my first screenshot. The Export button is there, but the file dialogue (to select, where to save the resulting file) does not open.
kniffte

**Comment by st-schilling** [comment]
@cleidigh ,
correct, I'm configuring a search by date, with date before/after. The search works, but then the export button does not open the file dialogue.
I wouldn't know any ui options other than the search options (which aren't strictly ui options...).

**Comment by cleidigh** [comment]
hmm, I also just reconfirmed that 14.1.2 already has the filepicker argument fix.
So there are two differences in what you and I are seeing.
I can add some debug, but I don't know what is going on that we such different things. As I said I don't even have an export button to work with.

**Comment by cleidigh** [comment]
@st-schilling 
What os are you on?

**Comment by cleidigh** [comment]
ok I actually found the filepicker issue in a second call.
I can fix that and add some debug.
Still have to get my ui to work.

**Comment by cleidigh** [comment]
ok try this b4 beta
install with the install addon from file
https://github.com/thunderbird/import-export-tools-ng/blob/v14.1.3/xpi/beta/import-export-tools-ng-14.1.3-b4-tb.xpi

**Comment by st-schilling** [comment]
Hello @cleidigh ,

great - that works :)
I was able to search for the email and upon clicking "Export", the file dialogue opened and the export was done.

Thanks a lot and have a good night.
kniffte

**Comment by st-schilling** [comment]
> @st-schilling What os are you on?

KUbuntu - Ubuntu 24.04.1 LTS

**Comment by cleidigh** [comment]
@st-schilling 
Great.
I also figured out my ui issue although I am not sure why you don't have the issue.
Note: b4 has some partial work for plaintext output.
@cleidigh

**Comment by marusan0303** [comment]
cleidigh, you are truly amazing.
I really appreciate it.
The issue is resolved.
Thank you very much.

**Comment by kiki-ja** [comment]
**ImportExportTools NG 14.1.2, 14.1.3-b3
Thunderbird 128.4.2esr 64bit
Windows 10 Pro 22H2 64bit**

**Reproduction Procedure:**
１. Right-click on the folder.
２. Select “ImportExportTools NG” from the context menu.
３. Click “Search And Export Messages” from the context menu of ImportExportTools NG.
４. The “Search Messages” dialog box opens.
５. Set the “Match” condition, specify “Format” = “Spreadsheet (CSV)”, and click “Export.
６. The CSV file is exported to the export folder you set (not an 'Excel' file).

![IETNG - Search Messages-01](https://github.com/user-attachments/assets/0f5a0ea6-efdd-4850-8426-bf45bdb7193a)

**Cause of no response and no export:**
You have not set the export folder in the ImportExportTools NG options.

![IETNG - Export directories-01](https://github.com/user-attachments/assets/c140e51b-6661-4020-b11e-a03f095b89ba)

@marusan0303
@st-schilling 
Did you set up an export folder?

ーーーーーーーーー

ImportExportTools NG 14.1.3-b4 has been improved so that any folder can be selected even if no export folder is set.
Thanks to @cleidigh.

@kiki-ja
